DataManagementSystem/Utilities/DMSHelpers.py

```diff
@@ -0,0 +1,235 @@
+        if candidateSEs:
+          siteSEMapping[self.local].setdefault( site, set() ).update( candidateSEs )
+          storageElementSet.update( candidateSEs )
+        else:
+          gLogger.debug( 'No SEs defined for site %s' % site )
+
+    # Add Sites from the SiteSEMappingByProtocol in the CS
+    siteSEMapping[self.protocol] = {}
+    cfgLocalSEPath = cfgPath( 'SiteSEMappingByProtocol' )
+    result = self.__opsHelper.getOptionsDict( cfgLocalSEPath )
+    if result['OK']:
+      sites = result['Value']
+      for site in sites:
+        candidates = self.__opsHelper.getValue( cfgPath( cfgLocalSEPath, site ), [] )
+        ses = set( candidates )
+        # If a candidate is a site, then all local SEs are eligible
+        for candidate in ses & siteSet:
+          ses.remove( candidate )
+          ses.update( siteSEMapping[self.local][candidate] )
+        siteSEMapping[self.protocol].setdefault( site, set() ).update( ses )
+
+    # Add Sites from the SiteSEMappingByDownload in the CS, else SiteLocalSEMapping (old convention)
+    siteSEMapping[self.download] = {}
+    cfgLocalSEPath = cfgPath( 'SiteSEMappingByDownload' )
+    result = self.__opsHelper.getOptionsDict( cfgLocalSEPath )
+    if not result['OK']:
+      cfgLocalSEPath = cfgPath( 'SiteLocalSEMapping' )
+      result = self.__opsHelper.getOptionsDict( cfgLocalSEPath )
+    if result['OK']:
+      sites = result['Value']
+      for site in sites:
+        candidates = self.__opsHelper.getValue( cfgPath( cfgLocalSEPath, site ), [] )
+        ses = set( candidates )
+        for candidate in ses & siteSet:
+          ses.remove( candidate )
+          ses.update( siteSEMapping[self.local][candidate] )
+        siteSEMapping[ self.download].setdefault( site, set() ).update( ses )
+
+    self.siteSEMapping = siteSEMapping
+    self.storageElementSet = storageElementSet
+    self.siteSet = siteSet
+    return S_OK( siteSEMapping )
+
+
+  def isSEFailover( self, storageElement ):
+    self.getSiteSEMapping()
+    if storageElement not in self.storageElementSet:
+      return False
+    seList = resolveSEGroup( self.__opsHelper.getValue( 'DataManagement/SEsUsedForFailover', [] ) )
+    return storageElement in resolveSEGroup( seList )
+
+  def isSEForJobs( self, storageElement ):
+    self.getSiteSEMapping()
+    if storageElement not in self.storageElementSet:
+      return False
+    seList = resolveSEGroup( self.__opsHelper.getValue( 'DataManagement/SEsNotToBeUsedForJobs', [] ) )
+    return storageElement not in resolveSEGroup( seList )
+
+  def isSEArchive( self, storageElement ):
+    self.getSiteSEMapping()
+    if storageElement not in self.storageElementSet:
+      return False
+    seList = resolveSEGroup( self.__opsHelper.getValue( 'DataManagement/SEsUsedForArchive', [] ) )
+    return storageElement in resolveSEGroup( seList )
+
+  def getSitesForSE( self, storageElement, connectionLevel = None ):
+    if not connectionLevel:
+      connectionLevel = self.download
```

Comment:
self.download is a pure constant, why it is used as a data member variable ? Should be rather 
    connectionLevel = DOWNLOAD
The same for other places and constants